src/local-history/local-history-manager.ts

```diff
@@ -33,6 +33,11 @@ export class LocalHistoryManager {
 
     constructor() {
         vscode.workspace.onDidSaveTextDocument((document: vscode.TextDocument) => {
+            if (this.checkRevisionLimit(document)) {
+                this.saveEditorContext(document);
+                this.limitRevisions(document);
```

Comment:
Shouldn't `limitRevisions` be called within `saveEditorContext`?
This way, if we were to programatically call `saveEditorContext` in other scenarios, the fact that we take into account the **limit** will already be handled.